src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -143,13 +145,7 @@ def _get_plate_reader_document_item(
 
     def _get_fluorescence_measurement_document(
         self, plate_block: PlateBlock, position: str
-    ) -> list[
-        Union[
-            UltravioletAbsorbancePointDetectionMeasurementDocumentItems,
-            FluorescencePointDetectionMeasurementDocumentItems,
-            LuminescencePointDetectionMeasurementDocumentItems,
-        ]
-    ]:
+    ) -> list[MeasurementDocumentItems]:
```

Comment:
This and all the other `_get_<measurement_document_type>_measurement_document` methods can only return one type of measurement, the signature should be consistent with that.

```suggestion
    ) -> list[FluorescencePointDetectionMeasurementDocumentItems]:
```

Reply:
The definition of measurement_document that is assigned with the result of this functions is MeasurementDocumentItems, so I can't define the return type as FluorescencePointDetectionMeasurementDocumentItems because I get 

**expression has type "list[FluorescencePointDetectionMeasurementDocumentItems]", variable has type "MeasurementDocumentItems"**

Reply:
@alejandro-salgado I pushed a change to the PR to fix this. By changing the typing of `measurement_document` to `Sequence` during dynamic assignment, and then casting it to a list, you can make the function signatures specific.